### ISSUE TITLE
channel_alias not working as advertised

### ISSUE THREAD

**Issue by ikalev** [conda/conda#2556]
``channel_alias`` does not appear to be doing anything:

```shell
$ set CHANNEL=http://myrepo/
$ conda config --set channel_alias %CHANNEL%
$ conda config --set channel_alias %CHANNEL% --system

$ conda install boost --channel dev
Fetching package metadata: ........Error: Could not find anaconda.org user dev
```

Tested with ``conda=4.0.6`` on Windows.

**Comment by bkreider** [comment]
Can you run `conda info` to make sure that conda is reading that setting correctly from your config file?

**Comment by ikalev** [comment]
``conda info`` returns:

```shell
$ conda info
Current conda install:

             platform : win-64
        conda version : 4.0.6
  conda-build version : 1.20.3
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : C:\ANACONDA  (writable)
  default environment : C:\ANACONDA
     envs directories : C:\ANACONDA\envs
        package cache : C:\ANACONDA\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : C:\Users\kalev\.condarc
    is foreign system : False
```

Contents of ``C:\Users\kalev\.condarc``:

```
$ type C:\Users\kalev\.condarc
channels:
  - defaults

ssl_verify: False
show_channel_urls: True
update_deps: True

channel_alias: http://myrepo
```

**Comment by bkreider** [comment]
Channel alias shouldn't be a valid user key/value.  I wonder if it is blocking the channel alias set in the system .condarc?  

Can you remove the key in your ~/.condarc and make sure it is in C:\ANACONDA\.condarc?  Can you post your centralized .condarc file?

**Comment by ikalev** [comment]
This is my ``--system`` ``.condarc``:

```
$ type c:\Anaconda\.condarc
ssl_verify: False

channels:
  - defaults

channel_alias: http://myrepo
```

It's identical to the user's ``.condarc`` (since I have added the alias to both files as shown earlier). Removing the alias from the user's config makes no difference. Still the same error.

**Comment by bkreider** [comment]
I think I've figured this bug out.  You have a few choices for a workaround right now (best to worst):
   - anaconda config --set url "http://myrepo"
   - Or add `add_anaconda_token: False`  to your .condarc.  
   - `conda remove anaconda-client`

Conda assumes (inconsistently) that if you have your anaconda-client pointed at anaconda.org, you probably want that for your channel_alias.  

@kalefranz

**Comment by ikalev** [comment]
Great. I am happy to wait for the next release. This should also resolve #1911 .

**Comment by kalefranz** [comment]
fixed with #2964
